app/controllers/question_answer/votes_controller.rb

```diff
@@ -138,5 +141,9 @@ def find_comment
     def ensure_can_see_comment!(comment)
       @guardian.ensure_can_see!(comment.post)
     end
+
+    def ensure_can_vote(author_id)
+      raise Discourse::InvalidAccess.new if author_id == current_user.id
```

Comment:
I don’t think this is sufficient, we also want to add a validation at the model level.

Reply:
Agreed, done.